// vanillajs-browser-helpers/index.cjs
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.scrollInfo = exports.saveFile = exports.replaceNode = exports.removeClass = exports.prepend = exports.position = exports.parseSelector = exports.once = exports.on = exports.off = exports.matches = exports.isWindow = exports.isEventTarget = exports.isDocument = exports.isDOMRoot = exports.isDOMNode = exports.isDOMFragment = exports.isDOMElement = exports.isDOMContainer = exports.isDOMChildNode = exports.isBlob = exports.invisible = exports.insertBefore = exports.insertAfter = exports.innerXML = exports.inView = exports.inDOM = exports.hidden = exports.hasClass = exports.getCurrentWindow = exports.getCurrentDocument = exports.findUniqueNodeCollection = exports.findByTagName = exports.findByQuery = exports.findByName = exports.findById = exports.findByClass = exports.find = exports.eventOptionsSupported = exports.ensureHTML = exports.elmIndex = exports.domReady = exports.delegate = exports.css = exports.create = exports.children = exports.boxModel = exports.attr = exports.append = exports.addClass = void 0;
exports.visible = exports.viewport = exports.vendorPrefixed = exports.trigger = exports.toggleClass = exports.toDOM = exports.supportsCSS = exports.size = exports.siblings = exports.selectorToHTML = exports.scrollParent = void 0;
var addClass_1 = require("./addClass");
Object.defineProperty(exports, "addClass", { enumerable: true, get: function () { return __importDefault(addClass_1).default; } });
var append_1 = require("./append");
Object.defineProperty(exports, "append", { enumerable: true, get: function () { return __importDefault(append_1).default; } });
var attr_1 = require("./attr");
Object.defineProperty(exports, "attr", { enumerable: true, get: function () { return __importDefault(attr_1).default; } });
var boxModel_1 = require("./boxModel");
Object.defineProperty(exports, "boxModel", { enumerable: true, get: function () { return __importDefault(boxModel_1).default; } });
var children_1 = require("./children");
Object.defineProperty(exports, "children", { enumerable: true, get: function () { return __importDefault(children_1).default; } });
var create_1 = require("./create");
Object.defineProperty(exports, "create", { enumerable: true, get: function () { return __importDefault(create_1).default; } });
var css_1 = require("./css");
Object.defineProperty(exports, "css", { enumerable: true, get: function () { return __importDefault(css_1).default; } });
var delegate_1 = require("./delegate");
Object.defineProperty(exports, "delegate", { enumerable: true, get: function () { return __importDefault(delegate_1).default; } });
var domReady_1 = require("./domReady");
Object.defineProperty(exports, "domReady", { enumerable: true, get: function () { return __importDefault(domReady_1).default; } });
var elmIndex_1 = require("./elmIndex");
Object.defineProperty(exports, "elmIndex", { enumerable: true, get: function () { return __importDefault(elmIndex_1).default; } });
var ensureHTML_1 = require("./ensureHTML");
Object.defineProperty(exports, "ensureHTML", { enumerable: true, get: function () { return __importDefault(ensureHTML_1).default; } });
var eventOptionsSupported_1 = require("./eventOptionsSupported");
Object.defineProperty(exports, "eventOptionsSupported", { enumerable: true, get: function () { return __importDefault(eventOptionsSupported_1).default; } });
var find_1 = require("./find");
Object.defineProperty(exports, "find", { enumerable: true, get: function () { return __importDefault(find_1).default; } });
var findByClass_1 = require("./findByClass");
Object.defineProperty(exports, "findByClass", { enumerable: true, get: function () { return __importDefault(findByClass_1).default; } });
var findById_1 = require("./findById");
Object.defineProperty(exports, "findById", { enumerable: true, get: function () { return __importDefault(findById_1).default; } });
var findByName_1 = require("./findByName");
Object.defineProperty(exports, "findByName", { enumerable: true, get: function () { return __importDefault(findByName_1).default; } });
var findByQuery_1 = require("./findByQuery");
Object.defineProperty(exports, "findByQuery", { enumerable: true, get: function () { return __importDefault(findByQuery_1).default; } });
var findByTagName_1 = require("./findByTagName");
Object.defineProperty(exports, "findByTagName", { enumerable: true, get: function () { return __importDefault(findByTagName_1).default; } });
var findUniqueNodeCollection_1 = require("./findUniqueNodeCollection");
Object.defineProperty(exports, "findUniqueNodeCollection", { enumerable: true, get: function () { return __importDefault(findUniqueNodeCollection_1).default; } });
var getCurrentDocument_1 = require("./getCurrentDocument");
Object.defineProperty(exports, "getCurrentDocument", { enumerable: true, get: function () { return __importDefault(getCurrentDocument_1).default; } });
var getCurrentWindow_1 = require("./getCurrentWindow");
Object.defineProperty(exports, "getCurrentWindow", { enumerable: true, get: function () { return __importDefault(getCurrentWindow_1).default; } });
var hasClass_1 = require("./hasClass");
Object.defineProperty(exports, "hasClass", { enumerable: true, get: function () { return __importDefault(hasClass_1).default; } });
var hidden_1 = require("./hidden");
Object.defineProperty(exports, "hidden", { enumerable: true, get: function () { return __importDefault(hidden_1).default; } });
var inDOM_1 = require("./inDOM");
Object.defineProperty(exports, "inDOM", { enumerable: true, get: function () { return __importDefault(inDOM_1).default; } });
var inView_1 = require("./inView");
Object.defineProperty(exports, "inView", { enumerable: true, get: function () { return __importDefault(inView_1).default; } });
var innerXML_1 = require("./innerXML");
Object.defineProperty(exports, "innerXML", { enumerable: true, get: function () { return __importDefault(innerXML_1).default; } });
var insertAfter_1 = require("./insertAfter");
Object.defineProperty(exports, "insertAfter", { enumerable: true, get: function () { return __importDefault(insertAfter_1).default; } });
var insertBefore_1 = require("./insertBefore");
Object.defineProperty(exports, "insertBefore", { enumerable: true, get: function () { return __importDefault(insertBefore_1).default; } });
var invisible_1 = require("./invisible");
Object.defineProperty(exports, "invisible", { enumerable: true, get: function () { return __importDefault(invisible_1).default; } });
var isBlob_1 = require("./isBlob");
Object.defineProperty(exports, "isBlob", { enumerable: true, get: function () { return __importDefault(isBlob_1).default; } });
var isDOMChildNode_1 = require("./isDOMChildNode");
Object.defineProperty(exports, "isDOMChildNode", { enumerable: true, get: function () { return __importDefault(isDOMChildNode_1).default; } });
var isDOMContainer_1 = require("./isDOMContainer");
Object.defineProperty(exports, "isDOMContainer", { enumerable: true, get: function () { return __importDefault(isDOMContainer_1).default; } });
var isDOMElement_1 = require("./isDOMElement");
Object.defineProperty(exports, "isDOMElement", { enumerable: true, get: function () { return __importDefault(isDOMElement_1).default; } });
var isDOMFragment_1 = require("./isDOMFragment");
Object.defineProperty(exports, "isDOMFragment", { enumerable: true, get: function () { return __importDefault(isDOMFragment_1).default; } });
var isDOMNode_1 = require("./isDOMNode");
Object.defineProperty(exports, "isDOMNode", { enumerable: true, get: function () { return __importDefault(isDOMNode_1).default; } });
var isDOMRoot_1 = require("./isDOMRoot");
Object.defineProperty(exports, "isDOMRoot", { enumerable: true, get: function () { return __importDefault(isDOMRoot_1).default; } });
var isDocument_1 = require("./isDocument");
Object.defineProperty(exports, "isDocument", { enumerable: true, get: function () { return __importDefault(isDocument_1).default; } });
var isEventTarget_1 = require("./isEventTarget");
Object.defineProperty(exports, "isEventTarget", { enumerable: true, get: function () { return __importDefault(isEventTarget_1).default; } });
var isWindow_1 = require("./isWindow");
Object.defineProperty(exports, "isWindow", { enumerable: true, get: function () { return __importDefault(isWindow_1).default; } });
var matches_1 = require("./matches");
Object.defineProperty(exports, "matches", { enumerable: true, get: function () { return __importDefault(matches_1).default; } });
var off_1 = require("./off");
Object.defineProperty(exports, "off", { enumerable: true, get: function () { return __importDefault(off_1).default; } });
var on_1 = require("./on");
Object.defineProperty(exports, "on", { enumerable: true, get: function () { return __importDefault(on_1).default; } });
var once_1 = require("./once");
Object.defineProperty(exports, "once", { enumerable: true, get: function () { return __importDefault(once_1).default; } });
var parseSelector_1 = require("./parseSelector");
Object.defineProperty(exports, "parseSelector", { enumerable: true, get: function () { return __importDefault(parseSelector_1).default; } });
var position_1 = require("./position");
Object.defineProperty(exports, "position", { enumerable: true, get: function () { return __importDefault(position_1).default; } });
var prepend_1 = require("./prepend");
Object.defineProperty(exports, "prepend", { enumerable: true, get: function () { return __importDefault(prepend_1).default; } });
var removeClass_1 = require("./removeClass");
Object.defineProperty(exports, "removeClass", { enumerable: true, get: function () { return __importDefault(removeClass_1).default; } });
var replaceNode_1 = require("./replaceNode");
Object.defineProperty(exports, "replaceNode", { enumerable: true, get: function () { return __importDefault(replaceNode_1).default; } });
var saveFile_1 = require("./saveFile");
Object.defineProperty(exports, "saveFile", { enumerable: true, get: function () { return __importDefault(saveFile_1).default; } });
var scrollInfo_1 = require("./scrollInfo");
Object.defineProperty(exports, "scrollInfo", { enumerable: true, get: function () { return __importDefault(scrollInfo_1).default; } });
var scrollParent_1 = require("./scrollParent");
Object.defineProperty(exports, "scrollParent", { enumerable: true, get: function () { return __importDefault(scrollParent_1).default; } });
var selectorToHTML_1 = require("./selectorToHTML");
Object.defineProperty(exports, "selectorToHTML", { enumerable: true, get: function () { return __importDefault(selectorToHTML_1).default; } });
var siblings_1 = require("./siblings");
Object.defineProperty(exports, "siblings", { enumerable: true, get: function () { return __importDefault(siblings_1).default; } });
var size_1 = require("./size");
Object.defineProperty(exports, "size", { enumerable: true, get: function () { return __importDefault(size_1).default; } });
var supportsCSS_1 = require("./supportsCSS");
Object.defineProperty(exports, "supportsCSS", { enumerable: true, get: function () { return __importDefault(supportsCSS_1).default; } });
var toDOM_1 = require("./toDOM");
Object.defineProperty(exports, "toDOM", { enumerable: true, get: function () { return __importDefault(toDOM_1).default; } });
var toggleClass_1 = require("./toggleClass");
Object.defineProperty(exports, "toggleClass", { enumerable: true, get: function () { return __importDefault(toggleClass_1).default; } });
var trigger_1 = require("./trigger");
Object.defineProperty(exports, "trigger", { enumerable: true, get: function () { return __importDefault(trigger_1).default; } });
var vendorPrefixed_1 = require("./vendorPrefixed");
Object.defineProperty(exports, "vendorPrefixed", { enumerable: true, get: function () { return __importDefault(vendorPrefixed_1).default; } });
var viewport_1 = require("./viewport");
Object.defineProperty(exports, "viewport", { enumerable: true, get: function () { return __importDefault(viewport_1).default; } });
var visible_1 = require("./visible");
Object.defineProperty(exports, "visible", { enumerable: true, get: function () { return __importDefault(visible_1).default; } });
